source/magpie/operation.d

```diff
@@ -203,6 +204,72 @@ auto merge(JoinTypes type = Inner, T, U)(T df1, U df2, string lsuffix = "x_", st
     else static assert(0, "Invalid join type. Available join type: left, right, outer, inner(default)");
 }
 
+enum AggregateOP
```

Comment:
I would strongly suggest to make this not a restricted set of ops e.g. this misses variance.

`auto aggregate(T, Ops...)(T df)` and then use with

```
auto x = df.aggregate!(min, max);
```

this will simplify the implementation to 
```
static foreach(i;0 ..  Ops.length)
{
    static foreach(j; 0 .. df.RowType.length)
         opres = Ops[i](df.data[j]);
         ret.indx.row.index[0][i] = Ops[i].stringof;
}
```
note that you still do `i` passes through the data, whereas you should be able to do all aggregation statistics with one pass. You should look at [fold](https://dlang.org/phobos/std_algorithm_iteration.html#.fold) and take some inspiration from there, particularly the example `arr.fold!(min, max)`.